### DISCUSSION TITLE
Experiment plan

### DISCUSSION THREAD

**Discussion by eddogola** [broadinstitute/superurop-log#5] [Ideas]
First run of experiments:

Big picture question - How much efficiency can we squeeze out of UNet and adjacent(MaskR-CNN+faster variants, SSD, MobileNet, FCN, FPN) models without losing on accuracy?

- [x] Experiment 1 - Learn/get a model on browser using ONNX [[yolov8 on onnx](https://github.com/broadinstitute/superurop-log/tree/main/experiments/yolov8)]
- [ ] Experiment 2 - Train, prune, quantize model from scratch
    - [x] U-Net [[U-Net from scratch](https://github.com/broadinstitute/superurop-log/tree/main/experiments/unet_train_prune)]
    - [ ] Squeeze U-Net [[Squeeze U-Net from scratch](https://github.com/broadinstitute/superurop-log/tree/main/experiments/squeeze-unet)]
    - [ ] SSD
- [ ] Experiment 3 - Fine-tune pre-trained models or ones with a pre-trained backbone
    - [x] U-Net (with pre-trained resnet50) [[U-Net with resnet50 backbone](https://github.com/broadinstitute/superurop-log/blob/main/experiments/unet_with_pretrained_resnet50_encoder.ipynb)]
    - [ ] MaskR-CNN** 
        - [x] [[RCNN on onnx](https://github.com/broadinstitute/superurop-log/tree/main/experiments/rcnn-onnx)]
    - [ ] SSD
    - [ ] MobileNet
    - [x] Yolov8 [[yolov8 on onnx](https://github.com/broadinstitute/superurop-log/tree/main/experiments/yolov8)]
- [ ] Experiment 5 - Porting an implementation to C/C++ and running using webassembly
- [ ] Experiment 4 - Compare efficiency, accuracy among models, and model limits(image size, throughput)
